src/Namotion.Interceptor.Registry/Paths/PathExtensions.cs

```diff
@@ -0,0 +1,261 @@
+            {
+                yield return property;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Tries to build a path from the root to the given property.
+    /// </summary>
+    /// <param name="pathProvider">The path provider to use.</param>
+    /// <param name="property">The property to build a path to.</param>
+    /// <param name="rootSubject">The root subject (optional, defaults to traversing to root).</param>
+    /// <returns>The path string, or null if unable to build path.</returns>
+    public static string? TryGetPath(
+        this PathProviderBase pathProvider,
+        RegisteredSubjectProperty property,
+        IInterceptorSubject? rootSubject = null)
+    {
+        var pathParts = new List<(RegisteredSubjectProperty property, object? index)>();
+        var current = property;
+        var registry = property.Subject.Context.TryGetService<ISubjectRegistry>();
+
+        if (registry is null)
+        {
+            return null;
+        }
+
+        while (current is not null)
+        {
+            pathParts.Add((current, null));
+
+            if (rootSubject is not null && current.Subject == rootSubject)
+            {
+                break;
+            }
+
+            // Find parent property that references this subject
+            var parentInfo = FindParentProperty(registry, current.Parent);
+            if (parentInfo is null)
+            {
+                break;
+            }
+
+            var (parentProperty, index) = parentInfo.Value;
+            if (pathParts.Count > 0)
+            {
+                // Update the last item with the index from parent
+                pathParts[^1] = (pathParts[^1].property, index);
+            }
+
+            current = parentProperty;
+        }
+
+        pathParts.Reverse();
+        return pathProvider.GetPath(pathParts);
+    }
+
+    private static IInterceptorSubject? GetChildSubject(RegisteredSubjectProperty property, object? index)
+    {
+        var value = property.GetValue();
+        if (value is null)
+        {
+            return null;
+        }
+
+        if (index is null)
+        {
+            return value as IInterceptorSubject;
+        }
+
+        if (property.IsSubjectDictionary && value is IDictionary dictionary)
+        {
+            return dictionary[index] as IInterceptorSubject;
+        }
+
+        if (property.IsSubjectCollection && value is IList list && index is int intIndex)
+        {
+            return list[intIndex] as IInterceptorSubject;
+        }
+
+        return null;
+    }
+
+    private static (RegisteredSubjectProperty property, object? index)? FindParentProperty(
+        ISubjectRegistry registry,
+        RegisteredSubject subject)
+    {
+        foreach (var parent in subject.Parents)
+        {
+            var parentSubject = registry.TryGetRegisteredSubject(parent.Property.Subject);
+
+            var parentProperty = parentSubject?.TryGetProperty(parent.Property.Name);
```

Comment:
here you can just use `var parentProperty = parent.Property?.TryGetRegisteredProperty();` (or similar) - dont need this complicated subject read first